.travis.yml

```diff
@@ -2,31 +2,29 @@
 sudo: false
 language: python
 python:
-  - "3.5"
-env:
-  - TESTENV=pypy-3.0.0-1.10-sqlite_file
-  - TESTENV=pypy3-2.9.2-1.8-sqlite_file
-  - TESTENV=pypy3-3.0.0-1.8-sqlite
-  - TESTENV=pypy3-3.0.0-1.8-sqlite_file
-  - TESTENV=python2.7-3.0.0-1.10-mysql_innodb
-  - TESTENV=python2.7-3.0.0-1.10-mysql_myisam
-  - TESTENV=python2.7-3.0.0-1.10-postgres
-  - TESTENV=python2.7-3.0.0-1.7-postgres
-  - TESTENV=python2.7-3.0.0-1.8-postgres
-  - TESTENV=python2.7-3.0.0-1.9-postgres
-  - TESTENV=python2.7-3.0.0-master-postgres
-  - TESTENV=python3.3-3.0.0-1.8-postgres
-  - TESTENV=python3.4-3.0.0-1.10-postgres
-  - TESTENV=python3.5-3.0.0-1.10-postgres
-  - TESTENV=python3.5-3.0.0-1.8-postgres
-  - TESTENV=python3.5-3.0.0-1.9-postgres
-  - TESTENV=python3.5-3.0.0-master-postgres
-  - TESTENV=checkqa-python2.7
-  - TESTENV=checkqa-python3.5
+  - 3.5
+  - 3.4
+  - 3.3
+  - 2.7
+  - pypy
+  - pypy3
+
 matrix:
+  fast_finish: true
+  include:
+    - python: 2.7
+      env: TOXENV=py27-pytest300-djangomaster-postgres
+    - python: 3.5
+      env: TOXENV=py35-pytest300-djangomaster-postgres
   allow_failures:
-    - env: TESTENV=python2.7-3.0.0-master-postgres
-    - env: TESTENV=python3.5-3.0.0-master-postgres
+    - env: TOXENV=py27-pytest300-djangomaster-postgres
+    - env: TOXENV=py35-pytest300-djangomaster-postgres
+
+cache:
+  directories:
+    - "${HOME}/virtualenv"
+    - "${TRAVIS_BUILD_DIR}/.tox"
```

Comment:
What about caching just `~/.cache/pip` instead?
After all the cache size is also an issue, since it will be fetched before the builds.

What is `${HOME}/virtualenv` used for?  Isn't that Travis' location for the Python?!